### ISSUE TITLE
Disaster Recovery DHCP / DNS

### ISSUE THREAD

**Issue by bagg3rs** [ministryofjustice/cloud-operations#47]
## User Story

As a CloudOps Engineer
I want to know how to recover the DHCP and DNS service
So that in the event of failure within these services I know what to do

## Value / Purpose

- Have clear guidance for new and existing team members and wider MoJ.

## Additional Information

Current [DHCP/DNS DR docs](https://github.com/ministryofjustice/staff-device-dns-dhcp-disaster-recovery)

## Definition of Done (DoD)

- Clear understanding of the DHCP/DNS recovery scenarios
- Our documentation is updated
- Additional tickets were created to document our other services even if it is to say `n/a`

Checklist for definition of done and acceptance criteria, example below (optional).

Example

- [ ] Documentation has been written / updated  
- [ ] README has been updated  
- [ ] Another team member has reviewed  

## Reference

[How to write good user stories](https://www.gov.uk/service-manual/agile-delivery/writing-user-stories)

**Comment by Gary-H9** [comment]
Hey team! Please [add your planning poker estimate](https://app.zenhub.com/workspaces/Cloud-Operations-Team-622a0b371800e400133bb924/issues/ministryofjustice/cloud-operations/47?planning-poker) with ZenHub @ASTRobinson @bagg3rs @kyphutruong @mitchdawson1982 @moontune @mTouhid @wanieldilson

**Comment by NeilRAdams** [comment]
Could you please include me in this review please? I need to understand the level & style of documentation being written in order to set expectations for other services down the line.
Also, I suggest we need to ensure that the IT Service Continuity folks in Service Operations (Mark Handford, Louse Stretch) are engaged as they play a governance role in assuring that 'adequate' documentation is in place. Therefore makes sense that we agreed what 'adequate' looks like and that can be reflected in the definition of done.

**Comment by wanieldilson** [comment]
Hi Neil, do we know what ServiceOps are expecting?  Is there a template/checklist etc we can start from?

**Comment by NeilRAdams** [comment]
Hi Dan. I don't think there is a template as such, but Mark Handford has shared the following.  I think it may be worth a conversation with that team, as some of the sections below fall into the service continuity arrangements (ensuring the thing has resilience in the first place) as opposed to Disaster Recovery plans which are focussed on picking it up if the worst does happen.
Looks like Mark is on long-term absence, but I can set up a session with colleagues in his team to agree an approach.
Can I ask if there are already DR plans in place for DHCP/DNS? If so it might be worth taking an approach of sharing what is already in place and then using that to identify any gaps. 
Have a look at the structure that Mark has previously provided and then we can have a chat about how this could progress.

 1.	Objective and Scope
2.	Brief overview of BAU service – which can hopefully be lifted from existing documentation 
3.	Business Criticality of the activity which this IT service supports – including BIA outcomes if one was performed – and any info you have on Business Continuity provisions that could be adopted if the IT service is unavailable - and how long they could be sustained 
4.	Resilience measures within the IT service –  any measures that have implemented to maintain availability and minimise the need to recover, including the elimination of SPOFs
5.	RTO and RPO 
6.	Envisaged Risk Scenarios and the strategy for recovering from each of them, including restoration priorities if appropriate.  
7.	Recovery Team and their roles – including comprehensive contact details (including for reserves)
8.	Process for DR invocation
9.	Resumption of BAU service from Recovery mode
10.	Plan Review – Suggest at least annual and following live/test invocations and significant changes to the functionality of the IT service
11.	Test Strategy – Approach, objectives, activities and reporting, including lessons learned and how they will be applied.  Should be at least annual.  Full test should precede live implementation of service or run ASAP afterwards 
12.	Any supporting documentation, including detailed Recovery scripting.  I’d include these as Appendices/Annexes

**Comment by Gary-H9** [comment]
Hi @NeilRAdams, this Issue is currently defined as three points, and I'm conscious that fitting it all of the above will grow the scope and not fit the User Story - which is explicitly in reference to "a CloudOps Engineer" and not the wider business. 

With reference to your input, I think this warrants further discussion, which we as a team should have internally to establish: 
a) what falls within the scope of this Issue
b) what falls outwith the scope of this Issue
c) Any other Stories / Issues which need to be created in relation to your feedback above. 

In relation to no template existing that's fine, this is why we like User Stories with a clear Definition of Done, it will enable both ourselves and those liaising with the team to have clarity on what we want to achieve in a piece of work prior to starting it.

**Comment by wanieldilson** [comment]
Agree with Gary, I think we need some separate user stories to satisfy the requirements of other users.  :thumbs-up:

**Comment by NeilRAdams** [comment]
I agree that this including all of the elements on that list from Service Ops would prevent this from fitting into this user story.
Having said that, by the nature of completing this user story it will cover off 2 of the key elements of the list from Service Ops. Specifically;
 
6.	Envisaged Risk Scenarios and the strategy for recovering from each of them, including restoration priorities if appropriate.  
12.	Any supporting documentation, including detailed Recovery scripting. 

Once this issue/user story has been completed, that will then give us an example that can be used to establish whether any additional user stories are required for the other elements. ta
